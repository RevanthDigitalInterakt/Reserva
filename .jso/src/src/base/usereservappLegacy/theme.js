  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
