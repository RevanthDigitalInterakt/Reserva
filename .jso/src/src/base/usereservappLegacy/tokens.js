  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
