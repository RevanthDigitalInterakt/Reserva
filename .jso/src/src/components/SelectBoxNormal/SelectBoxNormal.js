  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.SelectBoxNormal = SelectBoxNormal;
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[1]);
  var _testProps = _$$_REQUIRE(_dependencyMap[2])(_$$_REQUIRE(_dependencyMap[3]));
  var _EventProvider = _$$_REQUIRE(_dependencyMap[2])(_$$_REQUIRE(_dependencyMap[4]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function SelectBoxNormal(_ref) {
    var installmentsNumber = _ref.installmentsNumber,
      installmentsPrice = _ref.installmentsPrice,
      isChecked = _ref.isChecked,
      _onPress = _ref.onPress,
      installmentsEqualPrime = _ref.installmentsEqualPrime,
      _ref$price = _ref.price,
      price = _ref$price === undefined ? 0 : _ref$price;
    var _useRemoteConfig = (0, _$$_REQUIRE(_dependencyMap[5]).useRemoteConfig)(),
      getString = _useRemoteConfig.getString;
    var typeInstallments = (0, _react.useMemo)(function () {
      return getString('installments_prime');
    }, [getString]);
    var showInstallments = typeInstallments !== 'hide_installments' && installmentsEqualPrime || !installmentsEqualPrime;
    var separator = _reactNative.StyleSheet.flatten([_$$_REQUIRE(_dependencyMap[6]).styles.separator, typeInstallments === 'hide_installments' ? _$$_REQUIRE(_dependencyMap[6]).styles.mDefaultSeparator : _$$_REQUIRE(_dependencyMap[6]).styles.mtSeparator]);
    var pricesWrapper = _reactNative.StyleSheet.flatten([_$$_REQUIRE(_dependencyMap[6]).styles.priceContainer, _$$_REQUIRE(_dependencyMap[6]).styles.marginText, typeInstallments === 'hide_installments' ? _$$_REQUIRE(_dependencyMap[6]).styles.mDefaultPrice : _$$_REQUIRE(_dependencyMap[6]).styles.mtPrice]);
    var containerSelectBoxes = _reactNative.StyleSheet.flatten([_$$_REQUIRE(_dependencyMap[6]).styles.checkBoxContainer, typeInstallments === 'hide_installments' && _$$_REQUIRE(_dependencyMap[6]).styles.minHeight]);
    var renderPrice = (0, _react.useCallback)(function () {
      if (typeInstallments === 'show_prime_equal_to_regular' && installmentsEqualPrime) {
        return (installmentsEqualPrime == null ? undefined : installmentsEqualPrime.value) || 0;
      }
      if (!showInstallments) {
        return price;
      }
      return installmentsPrice;
    }, [installmentsEqualPrime, installmentsPrice, price, showInstallments, typeInstallments]);
    var renderInstallments = (0, _react.useCallback)(function () {
      if (typeInstallments === 'show_prime_equal_to_regular' && installmentsEqualPrime) {
        return installmentsEqualPrime == null ? undefined : installmentsEqualPrime.number;
      }
      return installmentsNumber;
    }, [installmentsEqualPrime, installmentsNumber, typeInstallments]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.TouchableOpacity, Object.assign({
      style: _$$_REQUIRE(_dependencyMap[6]).styles.normalPrice,
      onPress: function onPress() {
        _EventProvider.default.logEvent('normal_price_box_click', {});
        _onPress('priceNormal');
      },
      disabled: isChecked
    }, (0, _testProps.default)('com.usereserva:id/select_box_price_normal'), {
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, {
        style: containerSelectBoxes,
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, {
          style: _$$_REQUIRE(_dependencyMap[6]).styles.normalCheckBox,
          children: isChecked && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, Object.assign({
            style: _$$_REQUIRE(_dependencyMap[6]).styles.normalCheckBoxFill
          }, (0, _testProps.default)('com.usereserva:id/select_box_price_normal_checked')))
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, {
          style: _$$_REQUIRE(_dependencyMap[6]).styles.priceDataWrapper,
          children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
            fontFamily: "reservaSansRegular",
            style: _$$_REQUIRE(_dependencyMap[6]).styles.normalTextGray,
            children: "Pre\xE7o Normal"
          }), showInstallments && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, {
            style: _$$_REQUIRE(_dependencyMap[6]).styles.priceContainer,
            children: [showInstallments && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
              fontFamily: "reservaSansRegular",
              style: _$$_REQUIRE(_dependencyMap[6]).styles.normalText,
              children: [renderInstallments(), "x", ' ']
            }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
              style: _$$_REQUIRE(_dependencyMap[6]).styles.integerPart,
              fontFamily: "reservaSansRegular",
              children: ["R$", ' ', `${(0, _$$_REQUIRE(_dependencyMap[9]).integerPart)(renderPrice())},`]
            }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
              style: _$$_REQUIRE(_dependencyMap[6]).styles.decimalPart,
              fontFamily: "reservaSansRegular",
              children: (0, _$$_REQUIRE(_dependencyMap[9]).decimalPart)(renderPrice())
            })]
          })]
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, {
          style: separator
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.View, {
          style: _$$_REQUIRE(_dependencyMap[6]).styles.priceDataWrapper,
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, {
            style: pricesWrapper,
            children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
              style: [_$$_REQUIRE(_dependencyMap[6]).styles.integerPart, _$$_REQUIRE(_dependencyMap[6]).styles.normalTextBlack],
              fontFamily: "reservaSansRegular",
              children: ["R$", ' ', `${(0, _$$_REQUIRE(_dependencyMap[9]).integerPart)(price)},`]
            }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
              style: _$$_REQUIRE(_dependencyMap[6]).styles.normalTextBlack,
              fontWeight: "bold",
              fontFamily: "reservaSansRegular",
              children: (0, _$$_REQUIRE(_dependencyMap[9]).decimalPart)(price)
            })]
          })
        })]
      })
    }));
  }
