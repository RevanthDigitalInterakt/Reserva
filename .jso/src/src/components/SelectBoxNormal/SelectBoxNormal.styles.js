  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.styles = undefined;
  var _reactNative = _$$_REQUIRE(_dependencyMap[0]);
  var smallDevices = _reactNative.Dimensions.get('screen').width <= 375;
  var styles = exports.styles = _reactNative.StyleSheet.create({
    checkBoxContainer: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingHorizontal: 16,
      borderWidth: 1,
      borderColor: _$$_REQUIRE(_dependencyMap[1]).COLORS.LIGHT_GRAY
    },
    minHeight: {
      minHeight: 72
    },
    integerPart: {
      fontWeight: '700',
      fontSize: 24,
      color: _$$_REQUIRE(_dependencyMap[1]).COLORS.LIGHT_GRAY
    },
    decimalPart: {
      fontWeight: '700',
      color: _$$_REQUIRE(_dependencyMap[1]).COLORS.LIGHT_GRAY
    },
    marginText: {
      marginRight: 50
    },
    normalTextGray: {
      fontWeight: '400',
      fontSize: 18,
      color: _$$_REQUIRE(_dependencyMap[1]).COLORS.LIGHT_GRAY
    },
    normalTextBlack: {
      color: _$$_REQUIRE(_dependencyMap[1]).COLORS.BLACK
    },
    mtSeparator: {
      marginTop: 30
    },
    mDefaultSeparator: {
      margin: 0
    },
    mtPrice: {
      marginTop: 24
    },
    mDefaultPrice: {
      margin: 0
    },
    normalPrice: {
      marginHorizontal: 16,
      marginVertical: 12
    },
    normalCheckBox: {
      alignItems: 'center',
      justifyContent: 'center',
      padding: 3,
      height: 22,
      width: 22,
      borderRadius: 11,
      borderWidth: 1,
      borderColor: _$$_REQUIRE(_dependencyMap[1]).COLORS.BLACK,
      marginRight: smallDevices ? 15 : 0
    },
    normalCheckBoxFill: {
      height: 16,
      width: 16,
      borderRadius: 8,
      backgroundColor: _$$_REQUIRE(_dependencyMap[1]).COLORS.BLACK
    },
    normalText: {
      fontWeight: '400',
      fontSize: 18,
      color: _$$_REQUIRE(_dependencyMap[1]).COLORS.LIGHT_GRAY
    },
    priceDataWrapper: {
      alignItems: 'center',
      justifyContent: 'flex-end',
      paddingVertical: 10
    },
    ml: {
      marginLeft: 16
    },
    priceContainer: {
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'baseline',
      marginTop: 4
    },
    separator: {
      alignSelf: 'center',
      width: 1,
      height: 30,
      backgroundColor: _$$_REQUIRE(_dependencyMap[1]).COLORS.LIGHT_GRAY,
      marginHorizontal: smallDevices ? 15 : 0
    }
  });
