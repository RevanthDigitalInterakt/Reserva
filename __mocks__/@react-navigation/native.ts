const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
