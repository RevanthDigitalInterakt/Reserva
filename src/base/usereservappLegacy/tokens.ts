export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
