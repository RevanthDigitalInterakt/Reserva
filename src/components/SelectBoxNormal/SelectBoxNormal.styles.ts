import { StyleSheet, Dimensions } from 'react-native';
import { COLORS } from '../../base/styles/colors';

const smallDevices = Dimensions.get('screen').width <= 375;

export const styles = StyleSheet.create({
  checkBoxContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 16,
    borderWidth: 1,
    borderColor: COLORS.LIGHT_GRAY,
  },
  integerPart: {
    fontWeight: '700',
    fontSize: 24,
    color: COLORS.LIGHT_GRAY,
  },
  decimalPart: {
    fontWeight: '700',
    color: COLORS.LIGHT_GRAY,
  },
  marginText: {
    marginTop: 24,
    marginRight: 50,
  },
  normalTextGray: {
    fontWeight: '400',
    fontSize: 18,
    color: COLORS.LIGHT_GRAY,
  },
  normalTextBlack: {
    color: COLORS.BLACK,
  },
  normalPrice: {
    marginHorizontal: 16,
    marginVertical: 12,
  },
  normalCheckBox: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 3,
    height: 22,
    width: 22,
    borderRadius: 11,
    borderWidth: 1,
    borderColor: COLORS.BLACK,
    marginRight: smallDevices ? 15 : 0,
  },
  normalCheckBoxFill: {
    height: 16,
    width: 16,
    borderRadius: 8,
    backgroundColor: COLORS.BLACK,
  },
  normalText: {
    fontWeight: '400',
    fontSize: 18,
    color: COLORS.LIGHT_GRAY,
  },
  priceDataWrapper: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingVertical: 10,
  },
  priceContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'baseline',
    marginTop: 4,
  },
  separator: {
    alignSelf: 'center',
    width: 1,
    height: 30,
    backgroundColor: COLORS.LIGHT_GRAY,
    marginTop: 30,
    marginHorizontal: smallDevices ? 15 : 0,
  },
});
