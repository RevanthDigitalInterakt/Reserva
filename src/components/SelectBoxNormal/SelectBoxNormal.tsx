import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import {
  Typography,
  decimalPart,
  integerPart,
} from '@usereservaapp/reserva-ui';
import { styles } from './SelectBoxNormal.styles';
import testProps from '../../utils/testProps';

export interface SelectBoxNormalProps {
  onPress: (option: string) => void;
  isChecked: boolean;
  installmentsNumber: number;
  installmentsPrice: number;
  price?: number;
}

export function SelectBoxNormal({
  installmentsNumber,
  installmentsPrice,
  isChecked,
  onPress,
  price = 0,
}: SelectBoxNormalProps) {
  return (
    <TouchableOpacity
      style={styles.normalPrice}
      onPress={() => onPress('priceNormal')}
      disabled={isChecked}
      {...testProps('com.usereserva:id/select_box_price_normal')}
    >
      <View style={styles.checkBoxContainer}>
        <View style={styles.normalCheckBox}>
          {isChecked && (
            <View
              style={styles.normalCheckBoxFill}
              {...testProps('com.usereserva:id/select_box_price_normal_checked')}
            />
          )}
        </View>

        <View style={styles.priceDataWrapper}>
          <Typography
            fontFamily="reservaSansRegular"
            style={styles.normalTextGray}
          >
            Pre√ßo Normal
          </Typography>

          <View style={styles.priceContainer}>
            <Typography
              fontFamily="reservaSansRegular"
              style={styles.normalText}
            >
              {installmentsNumber}
              x
              {' '}
            </Typography>
            <Typography
              style={styles.integerPart}
              fontFamily="reservaSansRegular"
            >
              R$
              {' '}
              {`${integerPart(installmentsPrice)},`}
            </Typography>
            <Typography
              style={styles.decimalPart}
              fontFamily="reservaSansRegular"
            >
              {decimalPart(installmentsPrice)}
            </Typography>
          </View>
        </View>

        <View style={styles.separator} />

        <View style={styles.priceDataWrapper}>
          <View style={[styles.priceContainer, styles.marginText]}>
            <Typography
              style={[styles.integerPart, styles.normalTextBlack]}
              fontFamily="reservaSansRegular"
            >
              R$
              {' '}
              {`${integerPart(price)},`}
            </Typography>
            <Typography
              style={styles.normalTextBlack}
              fontWeight="bold"
              fontFamily="reservaSansRegular"
            >
              {decimalPart(price)}
            </Typography>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
}
