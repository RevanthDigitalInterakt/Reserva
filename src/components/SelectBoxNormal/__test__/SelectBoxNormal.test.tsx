import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '../../../base/usereservappLegacy/theme';
import { SelectBoxNormal } from '../SelectBoxNormal';

const onPressFn = jest.fn();

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <SelectBoxNormal
      installmentsNumber={4}
      installmentsPrice={64.5}
      isChecked={false}
      onPress={onPressFn}
      price={149.9}
    />
  </ThemeProvider>
);

describe('SelectBoxNormal', () => {
  beforeEach(() => render(TestingComponent));

  it('should render properly', () => {
    const selectBoxNormal = screen.getByTestId('com.usereserva:id/select_box_price_normal');

    expect(selectBoxNormal).toBeOnTheScreen();
  });

  it('should match with the snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call onPressFn when select box is pressed', () => {
    const selectBoxNormal = screen.getByTestId('com.usereserva:id/select_box_price_normal');

    fireEvent.press(selectBoxNormal);

    expect(onPressFn).toHaveBeenCalled();
    expect(onPressFn).toHaveBeenCalledWith('priceNormal');
  });
});
