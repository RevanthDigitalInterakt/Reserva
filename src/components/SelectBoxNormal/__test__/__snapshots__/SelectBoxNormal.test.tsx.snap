// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBoxNormal should match with the snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/select_box_price_normal"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 16,
      "marginVertical": 12,
      "opacity": 1,
    }
  }
  testID="com.usereserva:id/select_box_price_normal"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#8D8D8D",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#000000",
          "borderRadius": 11,
          "borderWidth": 1,
          "height": 22,
          "justifyContent": "center",
          "padding": 3,
          "width": 22,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
          "paddingVertical": 10,
        }
      }
    >
      <Text
        fontFamily="reservaSansRegular"
        style={
          Array [
            Object {
              "fontFamily": "ReservaSans-Regular",
            },
            Object {
              "color": "#8D8D8D",
              "fontSize": 18,
              "fontWeight": "400",
            },
          ]
        }
      >
        Pre√ßo Normal
      </Text>
      <View
        style={
          Object {
            "alignItems": "baseline",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginTop": 4,
          }
        }
      >
        <Text
          fontFamily="reservaSansRegular"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Regular",
              },
              Object {
                "color": "#8D8D8D",
                "fontSize": 18,
                "fontWeight": "400",
              },
            ]
          }
        >
          4
          x
           
        </Text>
        <Text
          fontFamily="reservaSansRegular"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Regular",
              },
              Object {
                "color": "#8D8D8D",
                "fontSize": 24,
                "fontWeight": "700",
              },
            ]
          }
        >
          R$
           
          64,
        </Text>
        <Text
          fontFamily="reservaSansRegular"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Regular",
              },
              Object {
                "color": "#8D8D8D",
                "fontWeight": "700",
              },
            ]
          }
        >
          50
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#8D8D8D",
          "height": 30,
          "marginTop": 30,
          "width": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "baseline",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 4,
            },
            Object {
              "marginRight": 50,
              "marginTop": 24,
            },
          ]
        }
      >
        <Text
          fontFamily="reservaSansRegular"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Regular",
              },
              Object {
                "color": "#8D8D8D",
                "fontSize": 24,
                "fontWeight": "700",
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          R$
           
          149,
        </Text>
        <Text
          fontFamily="reservaSansRegular"
          fontWeight="bold"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Regular",
                "fontWeight": "700",
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          90
        </Text>
      </View>
    </View>
  </View>
</View>
`;
